src/cfile/writer.py

```diff
@@ -432,7 +438,7 @@ def _write_typedef_usage(self, elem: core.TypeDef):
         """
         Writes typedef usage
         """
-        if not elem.name:
+        if elem.name == "":
```

Comment:
This is making it worse. `if not elem.name` works equally well if name is an empty string or None. Comparing it to empty string is not the Python way.